src/renderer/rendered_window.rs

```diff
@@ -225,26 +220,74 @@ impl RenderedWindow {
         }
 
         {
-            if 1.0 - self.scroll_t < std::f32::EPSILON {
-                // We are at destination, move t out of 0-1 range to stop the animation.
-                self.scroll_t = 2.0;
-                self.snapshots.clear();
+            let scroll_destination = 0.0;
+            let zeta = 1.0;
+            let omega = 4.0 / (zeta * settings.scroll_animation_length);
```

Comment:
Would it make sense to move this into the animation_utils.rs file? I think I get the basic gist of whats happening, but its not the clearest just inline like this without a label

Reply:
Just a suggestion though, take it or leave it

Reply:
Yes, conceptually it would belong better there. I have to think a little bit about the interface though.

And yes, you are also right I should have added some comments. I think I had comments at some point, but must have removed them in one of my countless re-bases. Or it might be in some parts that I did not include here, I had some improvements that uses linear regression to further smooth things out a tiny bit, but I determined that the improvements was so small, that it's not worth including at least not here.

The algorithm is just the `PD` part of the pretty standard https://en.wikipedia.org/wiki/PID_controller

And here's a nice presentation https://gdcvault.com/play/1027059/Math-In-Game-Development-Summit with an associated website to demonstrate it.

Reply:
The code has now been moved to `animation_utils.rs` and also documented.